checker/jdk/nullness/src/java/util/EnumMap.java

```diff
@@ -0,0 +1,839 @@
+    implements java.io.Serializable, Cloneable
+{
+    /**
+     * The <tt>Class</tt> object for the enum type of all the keys of this map.
+     *
+     * @serial
+     */
+    private final Class<K> keyType;
+
+    /**
+     * All of the values comprising K.  (Cached for performance.)
+     */
+    private transient K[] keyUniverse;
+
+    /**
+     * Array representation of this map.  The ith element is the value
+     * to which universe[i] is currently mapped, or null if it isn't
+     * mapped to anything, or NULL if it's mapped to null.
+     */
+    private transient @Nullable Object[] vals;
+
+    /**
+     * The number of mappings in this map.
+     */
+    private transient int size = 0;
+
+    /**
+     * Distinguished non-null value for representing null values.
+     */
+    private static final Object NULL = new Object() {
+        public int hashCode() {
+            return 0;
+        }
+
+        public String toString() {
+            return "java.util.EnumMap.NULL";
+        }
+    };
+
+    private Object maskNull(@Nullable Object value) {
+        return (value == null ? NULL : value);
+    }
+
+    @SuppressWarnings("unchecked")
+    private @Nullable V unmaskNull(@Nullable Object value) {
+        return (V)(value == NULL ? null : value);
+    }
+
+    private static final Enum<?>[] ZERO_LENGTH_ENUM_ARRAY = new Enum<?>[0];
+
+    /**
+     * Creates an empty enum map with the specified key type.
+     *
+     * @param keyType the class object of the key type for this enum map
+     * @throws NullPointerException if <tt>keyType</tt> is null
+     */
+    public EnumMap(Class<K> keyType) {
+        this.keyType = keyType;
+        keyUniverse = getKeyUniverse(keyType);
+        vals = new Object[keyUniverse.length];
+    }
+
+    /**
+     * Creates an enum map with the same key type as the specified enum
+     * map, initially containing the same mappings (if any).
+     *
+     * @param m the enum map from which to initialize this enum map
+     * @throws NullPointerException if <tt>m</tt> is null
+     */
+    public EnumMap(EnumMap<K, ? extends V> m) {
+        keyType = m.keyType;
+        keyUniverse = m.keyUniverse;
+        vals = m.vals.clone();
+        size = m.size;
+    }
+
+    /**
+     * Creates an enum map initialized from the specified map.  If the
+     * specified map is an <tt>EnumMap</tt> instance, this constructor behaves
+     * identically to {@link #EnumMap(EnumMap)}.  Otherwise, the specified map
+     * must contain at least one mapping (in order to determine the new
+     * enum map's key type).
+     *
+     * @param m the map from which to initialize this enum map
+     * @throws IllegalArgumentException if <tt>m</tt> is not an
+     *     <tt>EnumMap</tt> instance and contains no mappings
+     * @throws NullPointerException if <tt>m</tt> is null
+     */
+    @SuppressWarnings("nullness:method.invocation.invalid") // All the fields required by putAll method
```

Comment:
In this case, I suggest that you annotate `putAll` to express its contract:
 * the receiver is `@UnknownInitialization`
 * `@RequiresNonNull` for some expressions

It's better to express (and check!) the contract than to write `@SuppressWarnings`.  In this case, I don't think that you need `@SuppressWarnings`.